base64pp/include/base64pp/base64pp.h

```diff
@@ -22,6 +22,18 @@ namespace base64pp
     //! @return a base64 string containing the encoded data.
     std::string BASE64PP_EXPORT encode(std::span<std::uint8_t const> const input);
 
+    //! @brief Overload of the encode function for strings. This converts

```

Comment:
It looks like a `std::string const &` overload isn't required, as the `std::string_view` overload can also be used for `std::string` parameters:

https://stackoverflow.com/a/63166248 (last paragraph of this answer)
https://en.cppreference.com/w/cpp/string/basic_string/operator_basic_string_view

Reply:
Good point, strings are also implicitly converted to spans. I'll remove that overload and add a couple of tests for the `string_view` coverload. Thanks for spotting :)

Reply:
No, strings are implicitly converted to `string_view`, not to `span`... But I guess you mean this.

Reply:
Yes, that's what I meant 😄

Reply:
I've added tests, removed the unnecessary `std::string` overload. Also, had to rename the overload to `encode_str(...)` as it causes an ambiguous call when you want to encode nothing `{}` (initialiser lists, yay).

Reply:
> Also, had to rename the overload to `encode_str(...)` as it causes an ambiguous call when you want to encode nothing `{}` (initialiser lists, yay).

I think this type of problem is not specific for `std::span` / `std::string_view`. It appears also for trivial cases:
```
void foo(double){}
void foo(int){}
int main() { foo({}); }
```
```
test.cpp: In function ‘int main()’:
test.cpp:3:20: error: call of overloaded ‘foo(<brace-enclosed initializer list>)’ is ambiguous
 int main() { foo({}); }
                    ^
test.cpp:1:6: note: candidate: void foo(double)
 void foo(double){}
      ^~~
test.cpp:2:6: note: candidate: void foo(int)
 void foo(int){}
      ^~~
```
I have no real opinion on this, but I guess that nothing is wrong with my overload of `foo()`.
Is it important to be able to call `base64pp::encode({})`?

Reply:
It would not be important if the API hadn't already provided that functionality. Since it would be a breaking change, it's better to just rename the overload and keep the behaviour.

Maybe in future releases I'll change this behaviour and bump the major version. I've got some plans for changing the implementation with ranges, but I'm waiting for wider compiler support first (clang seems to be missing quite a lot of it rn)

Reply:
When the ranges rework happens, I'll come back to this problem 😆